lib/API/DX/Device.cpp

```diff
@@ -521,51 +528,103 @@ class DXDevice : public offloadtest::Device {
     addUploadEndBarrier(IS, Destination, R.isReadWrite());
   }
 
+  UINT getNumTiles(std::optional<int> NumTiles, UINT64 Width) {
+    UINT Ret;
+    if (NumTiles.has_value())
+      Ret = static_cast<UINT>(*NumTiles);
+    else
+      // Map the entire buffer by computing how many 64KB tiles cover it
+      Ret = static_cast<UINT>(
+          (Width + D3D12_TILED_RESOURCE_TILE_SIZE_IN_BYTES - 1) /
```

Comment:
could this operation overflow? Why is width a uint64?

Reply:
could width or d3d12_tiled_resource_tile_size_in_bytes be uint32 max or half that amount; could this operation overflow after you change width to be a uint32?

Reply:
Yes, though for this API:
[UpdateTileMappings](https://learn.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12commandqueue-updatetilemappings)
The pRangeTileCounts param is a UINT. Its value in my code comes directly from counting the number of tiles. So I don't think this should ever be > uint32_max.